### ISSUE TITLE
unicode in shader

### ISSUE THREAD

**Issue by ombschervister** [vispy/vispy#1163]
If comments in glsl-code has unicode symbols then shader compilation error raises.
Can you enable remove_comments function in gloo.preprocessor.preprocess?

```
WARNING: Traceback (most recent call last):
  File "D:/work/rexlab/test.py", line 49, in <module>
    app.run()
  File "D:\work\vispy\vispy\app\_default_app.py", line 62, in run
    return default_app.run()
  File "D:\work\vispy\vispy\app\application.py", line 137, in run
    return self._backend._vispy_run()
  File "D:\work\vispy\vispy\app\backends\_qt.py", line 232, in _vispy_run
    return app.exec_()
  File "D:\work\vispy\vispy\app\backends\_qt.py", line 434, in event
    out = super(QtBaseCanvasBackend, self).event(ev)
  File "D:\work\vispy\vispy\app\backends\_qt.py", line 434, in event
    out = super(QtBaseCanvasBackend, self).event(ev)
  File "D:\work\vispy\vispy\app\backends\_qt.py", line 706, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "D:\work\vispy\vispy\util\event.py", line 455, in __call__
    self._invoke_callback(cb, event)
  File "D:\work\vispy\vispy\util\event.py", line 475, in _invoke_callback
    self, cb_event=(cb, event))
  << caught exception here: >>
  File "D:\work\vispy\vispy\util\event.py", line 471, in _invoke_callback
    cb(event)
  File "D:/work/rexlab/test.py", line 45, in on_draw
    self.program.draw('triangle_strip')
  File "D:\work\vispy\vispy\gloo\program.py", line 470, in draw
    canvas.context.flush_commands()
  File "D:\work\vispy\vispy\gloo\context.py", line 171, in flush_commands
    self.glir.flush(self.shared.parser)
  File "D:\work\vispy\vispy\gloo\glir.py", line 232, in flush
    self._shared.flush(parser)
  File "D:\work\vispy\vispy\gloo\glir.py", line 146, in flush
    parser.parse(self._filter(self.clear(), parser))
  File "D:\work\vispy\vispy\gloo\glir.py", line 458, in parse
    self._parse(command)
  File "D:\work\vispy\vispy\gloo\glir.py", line 436, in _parse
    ob.set_shaders(*args)
  File "D:\work\vispy\vispy\gloo\glir.py", line 631, in set_shaders
    (type_ + ' shader', errormsg))
RuntimeError: Shader compilation error in vertex shader:
    (0) : error C0000: syntax error, unexpected $end, expecting "::" at token "<EOF>"
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x423cc88L>> for DrawEvent
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x423cc88L>> repeat 2
```

Program with unicode in glsl:

``` python
# coding=utf-8
from vispy import gloo, app
from vispy.gloo import Program


vertex = """
    attribute vec4 color;
    attribute vec2 position;
    varying vec4 v_color;
    void main()
    {
        /* bla bla bla бла бла бла */
        gl_Position = vec4(position, 0.0, 1.0);
        v_color = color;
    }
"""

fragment = """
    varying vec4 v_color;
    void main()
    {
        gl_FragColor = v_color;
    }
"""


class Canvas(app.Canvas):
    def __init__(self):
        app.Canvas.__init__(self, size=(512, 512), title='Rotating quad',
                            keys='interactive')

        # Build program & data
        self.program = Program(vertex, fragment)
        self.program['color'] = [(1, 0, 0, 1), (0, 1, 0, 1),
                                 (0, 0, 1, 1), (1, 1, 0, 1)]
        self.program['position'] = [(-1, -1), (-1, +1),
                                    (+1, -1), (+1, +1)]

        gloo.set_viewport(0, 0, 512, 512)

        self.show()

    def on_draw(self, event):
        gloo.clear((0.0, 0.0, 0.0, 1.0))
        self.program.draw('triangle_strip')

if __name__ == '__main__':
    c = Canvas()
    app.run()
```

**Comment by queenish001** [comment]
@djhoese

**Comment by djhoese** [comment]
I was going to suggest making a PR but looks like you did that and then closed it in #1164. I will close this too. If someone needs this then please create a new pull request for it to be discussed.
